Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManagerTest.java

```diff
@@ -0,0 +1,86 @@
+package edu.asu.spring.quadriga.service.impl.workbench;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.management.Query;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.spring.quadriga.dao.workbench.IRetrieveProjectDAO;
+import edu.asu.spring.quadriga.domain.proxy.ProjectProxy;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.dto.ProjectDTO;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workbench.mapper.IProjectShallowMapper;
+
+public class RetrieveProjectManagerTest {
+
+	@InjectMocks
+	private RetrieveProjectManager retrieveProjectManagerUnderTest;
+
+	@Mock
+	private IRetrieveProjectDAO mockedDBConnect = Mockito.mock(IRetrieveProjectDAO.class);
+
+	@Mock
+	private IProjectShallowMapper mockedProjectShallowMapper = Mockito.mock(IProjectShallowMapper.class);
+
+	@Before
+	public void setUp() throws QuadrigaStorageException {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void checkOnlyPublicProjectsAreReturned() throws QuadrigaStorageException{
+		Mockito.when(mockedDBConnect.getAllProjectsDTOByAccessibility("PUBLIC")).thenReturn(createPublicProjectDTOList());
```

Comment:
you also want to make sure that this method is called. use mockito's verify method.